hs-bindgen/test/internal/test-internal.hs

```diff
@@ -43,166 +48,177 @@ main :: IO ()
+        , ("typedef_vs_macro"            , defaultTracePredicate)
+        , ("typedefs"                    , defaultTracePredicate)
+        , ("typenames"                   , defaultTracePredicate)
+        , ("unions"                      , defaultTracePredicate)
+        , ("unnamed-struct"              , customTracePredicate $ \case
+            (TraceDiagnostic x) | diagnosticCategoryText x == "Semantic Issue" -> Just Expected
+            _otherTrace -> Nothing
+          )
+        , ("uses_utf8"                   , defaultTracePredicate)
+        , ("varargs"                     , customTracePredicate $ \case
+            TraceFrontend (FrontendParse (UnsupportedType _ UnsupportedVariadicFunction)) -> Just Expected
+            _otherTrace -> Nothing
+          )
+        , ("vector"                      , defaultTracePredicate)
         ]
     -- @rs-bindgen@ panics on these
-    , testGroup "examples/golden-norust" $ map goldenNoRust [
-          "macro_strings"
+    , testGroup "examples/golden-norust" $ map (uncurry goldenNoRust) [
+          ("macro_strings", defaultTracePredicate)
         ]
     , testGroup "examples/failing" [
           expectTrace
             "long_double"
-            "expected trace X"
-            (\case
-                TraceFrontend (FrontendParse (UnsupportedType _ UnsupportedLongDouble)) -> True
-                _otherwise -> False
+            (customTracePredicate $ \case
+              TraceFrontend (FrontendParse (UnsupportedType _ UnsupportedLongDouble)) -> Just Expected
+              _otherTrace -> Nothing
             )
         , expectTrace
             "implicit_fields_struct"
-            "expected trace UnsupportedImplicitFields"
-            (\case
-                TraceFrontend (FrontendParse (UnsupportedImplicitFields {})) -> True
-                _otherwise -> False
+            (customTracePredicate $ \case
+              TraceFrontend (FrontendParse (UnsupportedImplicitFields {})) -> Just Expected
+              _otherTrace -> Nothing
             )
         , expectTrace
             "declaration_unselected_b"
-            "expected trace MissingDeclaration"
-            (\case
-                TraceFrontend (FrontendNameMangler (MissingDeclaration {})) -> True
-                _otherwise -> False
+            (customTracePredicate $ \case
+              TraceFrontend (FrontendNameMangler (MissingDeclaration {})) -> Just Expected
+              _otherTrace -> Nothing
             )
         , expectTrace
             "redeclaration_different"
-            "expected trace Redeclaration"
-            (\case
-                TraceFrontend (FrontendSort (SortErrorDeclIndex (Redeclaration {}))) -> True
-                _otherwise -> False
+            (customTracePredicate $ \case
+              TraceFrontend (FrontendSort (SortErrorDeclIndex (Redeclaration {})))
+                -> Just Expected
+              TraceDiagnostic x |
```

Comment:
@edsko I had to add this expectation.

Reply:
Yeah, I guess that's reasonable enough. I wonder if we're going to check the clang diagnostic that we're getting here, perhaps we should lok for `redefined` in the diagnostic text rather than check the category..? But perhaps that will not work across all clang versions?

Reply:
Yes, I have found the `diagnosticCategoryText` to be the most stable, but we can try to be more specific in these cases. Should I check for a more detailed predicate?